flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -119,6 +122,9 @@ public boolean equals(Object obj) {
 		}
 		else if (obj != null && obj.getClass() == ResourceProfile.class) {
 			ResourceProfile that = (ResourceProfile) obj;
+			if (this == ResourceProfile.UNIVERSAL || that == ResourceProfile.UNIVERSAL) {
+				return true;
+			}
```

Comment:
The contract of the universal slot should not be reflected via the `equals` method of an `Object`. This is highly dangerous.